### PR TITLE
⚡ Write a deepgramLanguageDetection() Function using Deno #4465

### PR DESCRIPTION
**Pull request by Rinshin-Jalal**
## Description

According to the issue [#4465](https://github.com/appwrite/appwrite/issues/4465) on appwrite,

I created a Deno Cloud Function for detecting languages from audio using [Deepgram](https://deepgram.com/)

___

### 1. Success full response 
image

## 2. Error Response -  invalid API key

image

## 3. Error Response - invalid fileUrl
image

### CODE REVIEW

File: deno/deepgram-language-detection/README.md
Comment (gewenyu99) [inline deno/deepgram-language-detection/README.md:51]: Maybe add a -d here to run the docker command detached

Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/README.md:51]: Sorry I don't understand

File: deno/deepgram-language-detection/README.md
Comment (gewenyu99) [inline deno/deepgram-language-detection/README.md:37]: Please fix the instruction here

**Review by gewenyu99 — CHANGES_REQUESTED** [review]
Hi, I also could not build this demo. Following the instructions gave me the following error:
```
error: Module not found "file:///usr/builds/src/mod.ts".
```

Please update the directory in the instructions in your README and if the `mod.ts` error sounds familiar to you :)

File: deno/deepgram-language-detection/README.md
Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/README.md:37]: It is not a familiar error!
 let me check once more

Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/README.md:37]: @gewenyu99 
As you know
This command is to clone `open-runtimes/examples` not my fork ( It will only work after it is merged!!
So there is no deno/

I included this so that after the pull request was merged, maintainers wouldn't need to edit it. 

For now, clone this https://github.com/Rinshin-Jalal/open-runtimes-examples/
and change the branch to `deepgram-language-detection-deno`

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by Rinshin-Jalal** [comment]
> Thank you so much for the PR star_struck. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! jack_o_lantern

thanks I will be there for any changes

File: deno/deepgram-language-detection/README.md
Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md:51]: @gewenyu99 this might be fine considering most of the examples in the repo don't have the `-d` flag.

File: deno/deepgram-language-detection/README.md
Comment (gewenyu99) [reply deno/deepgram-language-detection/README.md:37]: Yes @Rinshin-Jalal, as @stnguyen90 mentioned, the instructions still do not match the actual folder name.

File: deno/deepgram-language-detection/README.md
Comment (gewenyu99) [reply deno/deepgram-language-detection/README.md:51]: @stnguyen90 👍 sure, I honestly think we should run all of them detached ;)

But I will accept this since all the other examples are like this.

**Review by gewenyu99 — CHANGES_REQUESTED** [review]

File: deno/deepgram-language-detection/src/mod.ts
Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/src/mod.ts:3]: Ok,

But I can't handle JSON because it is controlled by openruntime 
, All I can handle is `payload`, `variables`

**Review by gewenyu99 — CHANGES_REQUESTED** [review]
This looks good to me now, all we need is to update the example cURL command :)

File: deno/deepgram-language-detection/README.md
Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md:59]: @gewenyu99 @Rinshin-Jalal I would actually prefer this to be passed as a string rather than JSON object. That said, the code doesn't support the data passed as a string.

Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md:59]: @Rinshin-Jalal please make sure to [request a re-review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) after testing your code.

File: deno/deepgram-language-detection/src/mod.ts
Comment (stnguyen90) [reply deno/deepgram-language-detection/src/mod.ts:3]: Yes, the idea is to pass a JSON string and then decode `req.payload` in your function code. For example, see:

https://github.com/open-runtimes/examples/blob/d705ca6bc58f185b41d3f62da91280f582e7717f/deno/convert-phone-number-to-country-name/mod.ts#L24

Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/src/mod.ts:3]: When I use JSON.parse(req.payload)

It gives back an error 

Without it, code is working properly....

I asked about this before on the GitHub issue and they told me, you don't need to do that...

File: deno/deepgram-language-detection/README.md
Comment (Rinshin-Jalal) [reply deno/deepgram-language-detection/README.md:59]: > @gewenyu99 @Rinshin-Jalal I would actually prefer this to be passed as a string rather than JSON object to make it compatible with Appwrite Functions. That said, the code doesn't support the data passed as a string.

I can't handle that because it is handled by openruntimes itself we don't need too..

Data passed as string is also working

File: deno/deepgram-language-detection/README.md
Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md:51]: @gewenyu99, can this be resolved now?

File: deno/deepgram-language-detection/README.md
Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md]: @gewenyu99, can this be resolved now?

File: deno/deepgram-language-detection/README.md
Comment (stnguyen90) [reply deno/deepgram-language-detection/README.md:59]: @gewenyu99, can this be resolved now?

**Review by stnguyen90 — APPROVED** [review]

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Review by Meldiron — APPROVED** [review]

**Comment by Meldiron** [comment]
Thanks for the amazing PR 😇 I have tested it and all works great.
Merging the PR now 🥳

**Pull request merged**
